main.go

```diff
@@ -14,191 +16,328 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const concurrentFetch = 100
+
+// Commandline flags.
 var (
-	configFile  = flag.String("config.file", "", "Path to config file.")
-	metricsPath = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
+	addr           = flag.String("web.listen-address", ":9105", "Address to listen on for web interface and telemetry")
+	autoDiscover   = flag.Bool("exporter.discovery", false, "Discover all Mesos slaves")
+	localAddr      = flag.String("exporter.local-address", "127.0.0.1:5051", "Address to connect to the local Mesos slave")
+	masterURL      = flag.String("exporter.discovery.master", "http://mesos-master.example.com:5050", "Mesos master URL")
+	metricsPath    = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics")
+	scrapeInterval = flag.Duration("exporter.interval", (60 * time.Second), "Scrape interval duration")
 )
 
 var (
-	cpuLimitVal = prometheus.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Namespace: "mesos_task",
-			Name:      "cpu_limit",
-			Help:      "Fractional CPU limit.",
-		},
-		[]string{"task", "mesos_slave", "framework_id"},
+	variableLabels = []string{"task", "mesos_slave", "framework_id"}
+
+	cpuLimitDesc = prometheus.NewDesc(
+		prometheus.BuildFQName("", "mesos_task", "cpu_limit"),
```

Comment:
This works fine - it would be a bit more Prometheus-idiomatic to put the `mesos` part into the first (now empty) argument (the namespace). In the end, all the args will simply be glued together with underscores, so it will result in the same metric. Alternatively, one could also just not use `BuildFQName` at all and just say the whole metric name directly: `mesos_task_cpu_limit`. Especially since we don't use constants for the namespace and subsystem, that it probably the easiest and allows for nice greppability later :)